napari/layers/graph/graph.py

```diff
@@ -0,0 +1,322 @@
+        border_width=0.05,
+        border_width_is_relative=True,
+        border_color='dimgray',
+        border_color_cycle=None,
+        border_colormap='viridis',
+        border_contrast_limits=None,
+        face_color='white',
+        face_color_cycle=None,
+        face_colormap='viridis',
+        face_contrast_limits=None,
+        out_of_slice_display=False,
+        n_dimensional=None,
+        name=None,
+        metadata=None,
+        scale=None,
+        translate=None,
+        rotate=None,
+        shear=None,
+        affine=None,
+        opacity=1,
+        blending='translucent',
+        visible=True,
+        cache=True,
+        property_choices=None,
+        experimental_clipping_planes=None,
+        shading='none',
+        canvas_size_limits=(2, 10000),
+        antialiasing=1,
+        shown=True,
+    ) -> None:
+        if BaseGraph is None:
+            raise ImportError(
+                trans._(
+                    "`napari-graph` module is required by the graph layer."
+                )
+            )
+
+        self._data = self._fix_data(data, ndim)
+        self._edges_indices_view = []
+
+        super().__init__(
+            self._data,
+            ndim=self._data.ndim,
+            features=features,
+            feature_defaults=feature_defaults,
+            properties=properties,
+            text=text,
+            symbol=symbol,
+            size=size,
+            border_width=border_width,
+            border_width_is_relative=border_width_is_relative,
+            border_color=border_color,
+            border_color_cycle=border_color_cycle,
+            border_colormap=border_colormap,
+            border_contrast_limits=border_contrast_limits,
+            face_color=face_color,
+            face_color_cycle=face_color_cycle,
+            face_colormap=face_colormap,
+            face_contrast_limits=face_contrast_limits,
+            out_of_slice_display=out_of_slice_display,
+            n_dimensional=n_dimensional,
+            name=name,
+            metadata=metadata,
+            scale=scale,
+            translate=translate,
+            rotate=rotate,
+            shear=shear,
+            affine=affine,
+            opacity=opacity,
+            blending=blending,
+            visible=visible,
+            cache=cache,
+            property_choices=property_choices,
+            experimental_clipping_planes=experimental_clipping_planes,
+            shading=shading,
+            canvas_size_limits=canvas_size_limits,
+            antialiasing=antialiasing,
+            shown=shown,
+        )
+
+    @staticmethod
+    def _fix_data(
+        data: Optional[Union[BaseGraph, ArrayLike]] = None,
+        ndim: Optional[int] = None,
+    ) -> BaseGraph:
+        """Checks input data and return a empty graph if is None."""
+        if ndim is None:
+            ndim = 3
+
+        if data is None:
+            return UndirectedGraph(n_nodes=100, ndim=ndim, n_edges=200)
```

Comment:
The graph is empty. It pre-allocates the buffers with an arbitrary number of nodes and edges.
Maybe we could improve the `napari-graph` API because it doesn't look like it's an empty graph.
I'm adding a comment explaining this.

Reply:
Ah yeah, since you don't actually define any edges, then that makes sense. If these are sensible preallocation numbers, then maybe they should be the defaults for in napari-graph and this can just be `UndirectedGraph()`.

Reply:
Done, I forgot `napari-graph` already supports that.